### PR TITLE
chore: Centralise status management using collect status event

### PR DESCRIPTION
**Pull request by saltiyazan**
# Description

This PR changes the way we handle status management in the Lego charms, and it changes the expectations towards the charms implementing the lib:
- Centralise status management using collect status event
- Adds tests
- Updates documentation

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I Have bumped the version of the library

### CODE REVIEW

File: lib/charms/lego_base_k8s/v0/lego_client.py [hunk 1]
```diff
@@ -130,44 +123,96 @@ def __init__(self, *args, plugin: str):
...
+            return
+        if (err := self._validate_plugin_config()):
+            event.add_status(
+                BlockedStatus(err)
+            )
+            return
+        event.add_status(ActiveStatus())
+
+    def _sync_certificates(self, event: EventBase) -> None:
+        """Goes through all the certificates relations and handles outstanding requests."""
```
Comment (gruyaume) [inline lib/charms/lego_base_k8s/v0/lego_client.py:150]:
```suggestion
        """Go through all the certificates relations and handle outstanding requests."""
```

Comment (gruyaume) [reply lib/charms/lego_base_k8s/v0/lego_client.py:150]: Does the linter ignore this file? I believe it should catch this category of lints

File: lib/charms/lego_base_k8s/v0/lego_client.py [hunk 2]
```diff
@@ -130,44 +123,96 @@ def __init__(self, *args, plugin: str):
...
+                )
+
+    def _on_certificate_creation_request(self, event: CertificateCreationRequestEvent) -> None:
+        """Handle certificate creation request event.
+
+        - Retrieves subject from CSR
+        - Pushes CSR to workload container
+        - Executes lego command in workload
+        - Pulls certificates from workload
+        - Sends certificates to requesting charm
```
Comment (gruyaume) [inline lib/charms/lego_base_k8s/v0/lego_client.py:176]:
```suggestion
```

**Review by gruyaume — COMMENTED** [review]
Looks good, just a couple of nits

File: lib/charms/lego_base_k8s/v0/lego_client.py (hunk 1 above)
Comment (saltiyazan) [reply lib/charms/lego_base_k8s/v0/lego_client.py]: Seems like the choice of word here is unfortunate ("Goes"), the lint checker does indeed catch these errors, it didn't for this word. (tested a different word :D )
However I found other issues we had with the config of ruff and pyright, I fixed them, please take a look :)

**Review by gruyaume — APPROVED** [review]

**Pull request merged**